fake-snippets-api/routes/api/orders/get.ts

```diff
@@ -26,6 +49,90 @@ export default withRouteSpec({
     })
   }
 
+  if (_simulate_scenario) {
+    const errorMap = {
+      are_gerbers_generated: {
+        error_code: "GERBER_GENERATION_FAILED",
+        message: "Gerber generation failed",
+      },
+      are_gerbers_uploaded: {
+        error_code: "GERBER_UPLOAD_FAILED",
+        message: "Gerber upload failed",
+      },
+      is_gerber_analyzed: {
+        error_code: "ANALYZE_GERBER_FAILED",
+        message: "Analyze gerber failed",
+      },
+      is_bom_uploaded: {
+        error_code: "BOM_UPLOAD_FAILED",
+        message: "Bom upload failed",
+      },
+      is_pnp_uploaded: {
+        error_code: "PNP_UPLOAD_FAILED",
+        message: "Pnp upload failed",
+      },
+      is_bom_pnp_analyzed: {
+        error_code: "ANALYZE_BOM_AND_PNP_FAILED",
+        message: "Analyze bom and pnp failed",
+      },
+      is_bom_parsing_complete: {
+        error_code: "BOM_PARSING_FAILED",
+        message: "Bom parsing failed",
+      },
+      are_components_available: {
+        error_code: "COMPONENTS_AVAILABILITY_FAILED",
+        message: "Components availability failed",
+      },
+      is_patch_map_generated: {
+        error_code: "GENERATE_PATCH_MAP_FAILED",
+        message: "Generate patch map failed",
+      },
+      is_json_merge_file_created: {
+        error_code: "CREATION_JSON_MERGE_FILE_FAILED",
+        message: "Creation json merge file failed",
+      },
+      is_dfm_result_generated: {
+        error_code: "GENERATE_DFM_RESULT_FAILED",
+        message: "Generate dfm result failed",
+      },
+      are_files_downloaded: {
+        error_code: "DOWNLOAD_FILES_FAILED",
+        message: "Download files failed",
+      },
+      are_product_categories_fetched: {
+        error_code: "FETCH_PRODUCT_CATEGORIES_FAILED",
+        message: "Fetch product categories failed",
+      },
+      are_final_costs_calculated: {
+        error_code: "FINAL_COST_CALCULATION_FAILED",
+        message: "Final cost calculation failed",
+      },
+      is_json_merge_file_updated: {
+        error_code: "UPDATE_JSON_MERGE_FILE_FAILED",
+        message: "Update json merge file failed",
+      },
+      is_pcb_added_to_cart: {
+        error_code: "ADD_PCB_TO_CART_FAILED",
+        message: "Add pcb to cart failed",
+      },
+      is_added_to_cart: {
+        error_code: "ADD_TO_CART_FAILED",
+        message: "Add to cart failed",
+      },
+      are_initial_costs_calculated: {
+        error_code: "INITIAL_COST_CALCULATION_FAILED",
+        message: "Initial cost calculation failed",
+      },
+    }
+
+    const error = errorMap[_simulate_scenario]
+    if (error) {
+      order.error = errorSchema.parse(error)
+      order.has_error = true
+      ctx.db.updateOrder(order_id, { error: order.error, has_error: true })
```

Comment:
i think we still need binary flags on the order in order to display the checklist. You could put them inside an object, e.g. `order.jlcpcb_order_state = { are_gerbers_uploaded:false, ... }`

Reply:
If you look at the response being sent by this endpoint, the orderState has this data.